docker-compose.yml

```diff
@@ -338,6 +338,21 @@ services:
         - /var/run/docker.sock:/var/run/docker.sock:ro
         - /lib/modules:/lib/modules:ro
 
+
+    solr:
+      build: ./data/Dockerfiles/solr
+      restart: always
+      volumes:
+        - solr-vol-1:/opt/solr/server/solr/dovecot
```

Comment:
You're adding files to /opt/solr/server/solr/dovecot in the Dockerfile and then mount a volume on top. That will cause trouble, at least when updating the docker container in the future. I think you should mount to /opt/solr/server/solr/dovecot/data so that the volume only contains data and no config.

Also, you still have data/index/{segments_1,write.lock} checked in. Those will almost certainly be created automatically and shouldn't be in git.